include/ignition/rendering/ShaderParam.hh

```diff
@@ -64,29 +72,50 @@ namespace ignition
       /// \return Type of this parameter
       public: ParamType Type() const;
 
-      /// \brief Set from another ShaderParam.
-      /// \param[in] _other Another ShaderParam.
-      /// \return Reference to this ShaderParam.
+      /// \brief Get the element count of this parameter's buffer
+      /// \return Count of elements in this parameter's buffer
+      public: uint32_t Count() const;
+
+      /// \brief Set from another ShaderParam
+      /// \param[in] _other Another ShaderParam
+      /// \return Reference to this ShaderParam
       public: ShaderParam &operator=(const ShaderParam &_other);
 
-      /// \brief Set this to be a float param
-      /// \param[in] _value Value to set this param to.
+      /// \brief Set this to be a float parameter
+      /// \param[in] _value Value to set this parameter to
       public: void operator=(const float _value);
 
-      /// \brief Set this to be an integer param;
-      /// \param[in] _value Value to set this param to.
+      /// \brief Set this to be an integer parameter
+      /// \param[in] _value Value to set this parameter to
       public: void operator=(const int _value);
 
+      /// \brief Set this to be a word buffer parameter
+      /// \param[in] _count Number of elements in the word buffer
+      public: void InitializeWordBuffer(const uint32_t _count);
```

Comment:
we've started omitting `const` for primitive type arguments. There are other places in existing code that does this which we didn't clean up. But for new functions, we try to stick to the new convention.